### PR TITLE
GetAccountTransactions Pagination

### PR DESCRIPTION
**Pull request by nickaleks**
### Description of the Change

This PR introduces pagination logic to get account transactions query.
You can now specify starting hash and page size of the response.

### Benefits

Pagination

### Possible Drawbacks 

Slower queries due to added complexity

### Alternate Designs *[optional]*

In several places, additional indexes can be introduced in the future to optimize query

### CODE REVIEW

File: irohad/ametsuchi/impl/storage_impl.cpp [hunk 1]
```diff
@@ -606,9 +606,10 @@ CREATE TABLE IF NOT EXISTS account_has_grantable_permissions (
         + R"() NOT NULL,
     PRIMARY KEY (permittee_account_id, account_id)
 );
-CREATE TABLE IF NOT EXISTS height_by_hash (
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/storage_impl.cpp]: We still have that table in ametsuchi fixture. Replace it there as well

File: irohad/ametsuchi/impl/postgres_query_executor.cpp [hunk 2]
```diff
@@ -449,41 +450,100 @@ namespace iroha {
...
         const shared_model::interface::GetAccountTransactions &q) {
-      using QueryTuple =
-          QueryType<shared_model::interface::types::HeightType, uint64_t>;
+      using QueryTuple = QueryType<shared_model::interface::types::HeightType,
+                                   uint64_t,
+                                   uint64_t>;
       using PermissionTuple = boost::tuple<int>;
 
-      auto cmd = (boost::format(R"(WITH has_perms AS (%s),
+      auto &pagination_info = q.paginationMeta();
```
Comment (Akvinikym) [inline irohad/ametsuchi/impl/postgres_query_executor.cpp]: If `paginationMeta()` returns constref, it should be `const auto &pagination_info`

File: irohad/ametsuchi/impl/postgres_query_executor.cpp [hunk 3]
```diff
@@ -492,9 +552,27 @@ namespace iroha {
...
+              auto error = (boost::format("invalid pagination hash: %s")
+                            % first_hash->hex())
+                               .str();
+              return this->logAndReturnErrorResponse(
+                  QueryErrorType::kStatefulFailed, error);
+            }
 
-            return query_response_factory_->createTransactionsResponse(
-                std::move(response_txs), query_hash_);
+            // next transaction exists
```
Comment (Akvinikym) [inline irohad/ametsuchi/impl/postgres_query_executor.cpp]: Not very clear comment, please expand it a little

File: test/module/irohad/ametsuchi/postgres_query_executor_test.cpp [hunk 4]
```diff
@@ -259,10 +280,7 @@ namespace iroha {
                        .build();
       auto result = executeQuery(query);
       ASSERT_NO_THROW({
-        const auto &cast_resp = boost::apply_visitor(
-            framework::SpecifiedVisitor<
-                shared_model::interface::AccountResponse>(),
-            result->get());
+        const auto &cast_resp = getRef<AccountResponse>(result->get());
```
Comment (Akvinikym) [inline test/module/irohad/ametsuchi/postgres_query_executor_test.cpp]: Please, cancel changes in existing tests, as there will be a significant amount of conflicts with this [PR](https://github.com/hyperledger/iroha/pull/1901) when merging

**Pull request merged**